idaes/generic_models/unit_models/skeleton_model.py

```diff
@@ -0,0 +1,154 @@
+
+    def add_ports(self, name, member_dict, doc=None):
+        """
+        This is a method to build Port objects in the SkeletonUnitModel and
+        populate them with appropriate port members as specified. User can add
+        as many inlet and outlet ports as required.
+
+        Keyword Args:
+            name : name to use for Port object.
+            dict : dictionary containing variables to be added to the port
+            doc : doc string for Port object
+
+        Returns:
+            A Pyomo Port object and associated components.
+        """
+        # Validate that member_dict is a dict
+        if not isinstance(member_dict, dict):
+            raise ConfigurationError(
+                "member_dict should be a dictionary "
+                "with the keys being the name assigned(strings) and "
+                "values being the variable objects declared in the model.")
+
+        # Create empty Port
+        p = Port(noruleinit=True, doc=doc)
+
+        # Add port object to model
+        setattr(self, name, p)
+
+        # Populate port and map names to actual variables as defined
+        for k in member_dict.keys():
+            p.add(member_dict[k], name=k)
+
+    def _custom_initialize_method(self):
+        if self.custom_initializer_func is None:
+            raise ConfigurationError(
+                "_custom_initializer_func attribute was not set to a method. "
+                "Set a custom method before calling the initialize method.")
+        else:
+            self.custom_initializer_func
+
+    def initialize(self, outlvl=idaeslog.NOTSET,
+                   solver=None, optarg=None):
+        """Initialize method for the SkeletonUnitModel. If a custom initializer
+        is provided then, this method will use it. If no custom initializer is
+        provided, a final solve is executed after checking that degrees of
+        freedom is zero. This will be the only solve executed after checking
+        for zero degrees of freedom.
+
+        Args:
+            outlvl ([idaes logger], optional):
+                [Set idaes logger level]. Defaults to idaeslog.NOTSET.
+            solver ([string], optional):
+                [solver to be used for solve]. Defaults to None but if None,
+                ipopt from IDAES is used.
+            optarg ([dict], optional): [solver arguments]. Defaults to None
+                but if None, default args for ipopt from IDAES is used.
+
+        Raises:
+            ConfigurationError: [Checks for degrees of freedom before final
+            solve and raises exception when not zero.]
+        """
+        # Set solver options
+        init_log = idaeslog.getInitLogger(self.name, outlvl, tag="unit")
+        solve_log = idaeslog.getSolveLogger(self.name, outlvl, tag="unit")
+
+        opt = get_solver(solver=solver, options=optarg)
+
+        if self.config.custom_initializer:
+            self._custom_initialize_method()
+        elif degrees_of_freedom(self) == 0:
```

Comment:
Should this be a global check? I.e. shouldn't this also apply even if there is a custom initialisation method?